.github/workflows/cicd.yml

```diff
@@ -24,9 +24,11 @@ on:
     branches: [ main ]
 
 concurrency:
-  # Cancel any CI/CD workflow currently in progress for the same PR.
-  # Allow running concurrently with any other commits.
-  group: cicd-${{ github.event.pull_request.number || github.sha }}
+  # Cancel any in-progress instance of this workflow (CI/CD) for the same PR.
+  # Allow running concurrently with any commits on any other branch.
+  # Using GITHUB_REF instead of GITHUB_SHA allows parallel runs on
+  # different branches with the same HEAD commit.
+  group: cicd-${{ github.event.pull_request.number || github.ref }}
```

Comment:
GITHUB_SHA vs. GITHUB_REF:
https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables

_"The ref given is fully-formed, meaning that for branches the format is refs/heads/<branch_name>, for pull requests it is refs/pull/<pr_number>/merge, and for tags it is refs/tags/<tag_name>. For example, refs/heads/feature-branch-1."_

Reply:
GITHUN_SHA and GITHUB_REF values for workflow_dispatch event:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch